File: crates/normalize-facts/tests/fixtures/vb/add_numbers/project/MathUtils.vb
Module MathUtils

    Function Add(a As Integer, b As Integer) As Integer
        Return a + b
    End Function

    Function Multiply(a As Integer, b As Integer) As Integer
        Return a * b
    End Function

End Module

Class Calculator

    Private history As New List(Of Integer)

    Function Compute(op As String, a As Integer, b As Integer) As Integer
        Dim result As Integer
        If op = "add" Then
            result = Add(a, b)
        Else
            result = Multiply(a, b)
        End If
        history.Add(result)
        Return result
    End Function

End Class

File: crates/normalize-facts/tests/fixtures/cpp/add_numbers/project/math_utils.cpp
#include "math_utils.hpp"
#include <cstring>

namespace math {

int add(int a, int b) {
    return a + b;
}

int multiply(int a, int b) {
    return a * b;
}

Calculator::Calculator() : result_(0) {}

int Calculator::compute(const char *op, int a, int b) {
    if (std::strcmp(op, "add") == 0) {
        result_ = add(a, b);
    } else {
        result_ = multiply(a, b);
    }
    return result_;
}

int Calculator::last_result() const {
    return result_;
}

} // namespace math
